### ISSUE TITLE
Deprecated: array_key_exists() !! 

### ISSUE THREAD

**Issue by naamaan** [PoetOS/moodle-mod_questionnaire#299]
Hi, I'm using Moodle 3.8, I got the following error when using the plugin.

```
Deprecated: array_key_exists(): Using array_key_exists() on objects is deprecated. Use isset() or property_exists() instead in /var/www/html/moodle/mod/questionnaire/classes/question/question.php on line 897

Deprecated: array_key_exists(): Using array_key_exists() on objects is deprecated. Use isset() or property_exists() instead in /var/www/html/moodle/mod/questionnaire/classes/question/question.php on line 897
```
Any help?!

**Comment by sharpchi** [comment]
I'd say you're using a version of PHP that the plugin doesn't yet support. You're probably using PHP7.4. Though Moodle 3.8 supports PHP7.4 (not sure when that happened, I thought the limit was 7.2), not all plugins will have caught up yet.

**Comment by naamaan** [comment]
Thanks,
So will wait the update while it's not damaging the whole system.

But I've got new one:
`Notice: Trying to access array offset on value of type null in /var/www/html/moodle/mod/questionnaire/classes/event/course_module_viewed.php on line 54`

Is it related to the same problem?

**Comment by mchurchward** [comment]
What version of the questionnaire are you using? And can you provide steps as to how to reproduce the problem?

**Comment by naamaan** [comment]
Hi,
The version is: 
3.8.3 (Build - 2020062400)
2020011508

The steps are:
1- Add new Questionnaire activity anywhere in a course.
2- Press Save and display.
3- The error appears, please see the attached snap:

![image](https://user-images.githubusercontent.com/2176593/86090720-48b5ae80-bab3-11ea-92da-8b64b76fe466.png)

**Comment by mchurchward** [comment]
I think this is the same issue as https://github.com/PoetOS/moodle-mod_questionnaire/issues/291

**Comment by mchurchward** [comment]
The fix suggested in #291 should work. Pushed here b8137c0227407149d3f8f01c345548d13474feee.
